sdk/samples/iot/paho_iot_hub_c2d_sample.c

```diff
@@ -12,7 +12,7 @@
 // warning C4201: nonstandard extension used: nameless struct/union
 #pragma warning(disable : 4201)
 #endif
-#include <paho-mqtt/MQTTClient.h>
+#include <MQTTClient.h>
```

Comment:
This is a follow-up to https://github.com/Azure/azure-sdk-for-c/pull/2377#issuecomment-1299323996

I am assuming this is the only mitigation needed within the samples when upgrading the vcpkg baseline and the paho-mqtt dependency.